### ISSUE TITLE
[CLOSED] Wrap Finder formats in a Dropdown if GDS Editor

### ISSUE THREAD

**Issue by benilovj** [alphagov/specialist-publisher#299]
<a href="https://github.com/tommyp"><img src="https://avatars.githubusercontent.com/u/449004?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [tommyp](https://github.com/tommyp)**
_Wednesday 29 October 2014 at 17:44 GMT_
_Originally opened as https://github.com/alphagov/specialist-publisher/pull/299_

---

As GDS Editors, the amount of Finders was breaking the header navigation. Now, we wrap the Finders in dropdown tags if the user is from GDS. We also display the name of the current finder in the dropdown
if one is selected.

Also, adding a new route and permission check in the view requires adding a new item in the hash in the application controller.

![screen shot 2014-10-29 at 17 43 12 1](https://cloud.githubusercontent.com/assets/449004/4831245/190cd040-5f93-11e4-9146-cebcdae17104.png)

---

_**[tommyp](https://github.com/tommyp)** included the following code: https://github.com/alphagov/specialist-publisher/pull/299/commits_

**Comment by benilovj** [comment]
<a href="https://github.com/edds"><img src="https://avatars.githubusercontent.com/u/35035?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [edds](https://github.com/edds)**
_Thursday 30 October 2014 at 08:52 GMT_

---

Would it be possible to have the `finders` method be auto-populated from the schema files? 

Also some of your `document_type` values in the hash don't match the old values passed into `current_user_can_edit?` for example `cma_cases` rather than `cma_case`. Is that going to cause an issue?

**Comment by benilovj** [comment]
<a href="https://github.com/tommyp"><img src="https://avatars.githubusercontent.com/u/449004?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [tommyp](https://github.com/tommyp)**
_Thursday 30 October 2014 at 09:24 GMT_

---

Good catch on the `document_type`s. They should be singular. As for this being autopopulated, we probably could. We just have to add the `document_type` to the schema. Might require some refactoring though as the schemas are all separate objects at the moment.

**Comment by benilovj** [comment]
<a href="https://github.com/kalleth"><img src="https://avatars.githubusercontent.com/u/615627?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kalleth](https://github.com/kalleth)**
_Friday 31 October 2014 at 11:16 GMT_

---

@edds It should be possible to autopopulate these, yes, but I don't think the schema files is the right way to go about this.

We've got many different places within `specialist-publisher` that act as a "list" of finders and/or document types, among them:
- `config/routes.rb` (document_types = [])
- `app/views/layouts/application.html.erb` (this PR changes this list to `ApplicationController`)
- `ls app/controllers/`
- `ls schemas/`
- Distinct methods on the `RepositoryRegistry`
- ...and so on.

Basically these distinctions arise from wether we think the specification of each document format belongs in code or as configuration data. Historically we've chosen the former.

I think it would be a great idea to create a Single Point of Truth (tm) for which document formats the app is dealing with, but I think that would be best as a separate piece of work as to do it properly requires touching so many parts of the application.

**Comment by benilovj** [comment]
<a href="https://github.com/kalleth"><img src="https://avatars.githubusercontent.com/u/615627?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kalleth](https://github.com/kalleth)**
_Friday 31 October 2014 at 11:18 GMT_

---

:+1: This is an awesome change that will make our lives better (and hopefully not change things for people other than GDS editors).

**Comment by benilovj** [comment]
<a href="https://github.com/edds"><img src="https://avatars.githubusercontent.com/u/35035?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [edds](https://github.com/edds)**
_Friday 31 October 2014 at 11:29 GMT_

---

@kalleth that is fine. I was more intrigued if it would be easily possible. The `README` should probably be updated now as it talks about adding a link to `application.html.erb`.
